src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java

```diff
@@ -32,25 +34,60 @@ public PublicationService(ArtefactRepository artefactRepository,
 
     /**
      * Method that handles the creation or updating of a new publication.
+     *
      * @param artefact The artifact that needs to be created.
-     * @param payload The paylaod for the artefact that needs to be created.
+     * @param payload  The paylaod for the artefact that needs to be created.
      * @return Returns the UUID of the artefact that was created.
      */
     public Artefact createPublication(Artefact artefact, String payload) {
 
-        Optional<Artefact> foundArtefact =  artefactRepository
+        Optional<Artefact> foundArtefact = artefactRepository
             .findBySourceArtefactIdAndProvenance(artefact.getSourceArtefactId(), artefact.getProvenance());
 
         foundArtefact.ifPresent(value -> artefact.setArtefactId(value.getArtefactId()));
 
         String blobUrl = azureBlobService.createPayload(
             artefact.getSourceArtefactId(),
             artefact.getProvenance(),
-            payload);
+            payload
+        );
 
         artefact.setPayload(blobUrl);
         artefact.setSearch(payloadExtractor.extractSearchTerms(payload));
 
         return artefactRepository.save(artefact);
     }
+
+
+    /**
+     * Get all relevant artefacts relating to a given court ID.
+     *
+     * @param searchValue - represents the court ID in question being searched for
+     * @param verified    - represents the verification status of the user. Currently only verified/non-verified, but
+     *                    will include other verified user types in the future
+     * @return a list of all artefacts that fulfil the timing criteria, match the given court id and sensitivity
+     *                    associated with given verification status
+     */
+    public List<Artefact> findAllWithSearch(String searchValue, Boolean verified) {
+        LocalDateTime currDate = LocalDateTime.now();
+        if (verified) {
+            return artefactRepository.findArtefactsBySearchVerified(searchValue, currDate);
+        } else {
+            return artefactRepository.findArtefactsBySearchUnverified(searchValue, currDate);
+        }
+
+    }
+
+    /**
+     * takes in provenance and artefact id and returns the data within the matching blob in string format.
+     *
+     * @param provenance represents the provenance component of the blob
+     * @param artefactId represents the artefact id component of the blob
+     * @return the data within the blob in string format
+     */
+    public String getFromBlobStorage(String provenance, String artefactId) {
```

Comment:
Similar to the API comment around sourceArtefactID in the controller. I would rename this to 'getByArtefactId', as it then unlinks the calling classes (e.g controller) from the underlying repository (blob)

Reply:
Done.